src/native/public/mono/utils/mono-counters.h

```diff
@@ -1,5 +1,8 @@
 /**
  * \file
+ *
+ * WARNING: The functions in this header are no-ops and provided for source compatibility with older versions of mono only.
+ *
```

Comment:
I kept this since it was public API but given it never did anything in dotnet/runtime we could also remove it, depending on how important we deem source compatibility with mono/mono. Let me know what you think.

Reply:
This makes sense to me.  Maybe also add the same warning to`details/mono-counters-functions.h`